cluster/juju/layers/kubeapi-load-balancer/reactive/load_balancer.py

```diff
@@ -56,11 +57,16 @@
     endscript
 }"""
 
+cert_dir = Path('/srv/kubernetes/')
+server_crt_path = cert_dir / 'server.crt'
+server_key_path = cert_dir / 'server.key'
+
 
 @when('certificates.available', 'website.available')
-def request_server_certificates(tls, website):
+def request_server_certificates():
     '''Send the data that is required to create a server certificate for
     this server.'''
+    website = endpoint_from_flag('website.available')
```

Comment:
Similar to @hyperbolic2346's concern [here](https://github.com/juju-solutions/kubernetes/pull/207#pullrequestreview-194173248): what if website.available is not set when this is called from upgrade_charm?

Reply:
Yes, that needs to be handled.  I think it's best to handle these by making the calls conditional in the upgrade handler, rather than trying to handle them within this handler.

Reply:
Either way works for me :+1: